crates/rattler_conda_types/src/repo_data/mod.rs

```diff
@@ -457,33 +463,17 @@ fn determine_subdir(
     let platform = platform.ok_or(ConvertSubdirError::PlatformEmpty)?;
     let arch = arch.ok_or(ConvertSubdirError::ArchEmpty)?;
 
-    let plat = match platform.as_ref() {
-        "linux" => match arch.as_ref() {
-            "x86" => Ok(Platform::Linux32),
-            "x86_64" => Ok(Platform::Linux64),
-            "aarch64" => Ok(Platform::LinuxAarch64),
-            "armv61" => Ok(Platform::LinuxArmV6l),
-            "armv71" => Ok(Platform::LinuxArmV7l),
-            "ppc64le" => Ok(Platform::LinuxPpc64le),
-            "ppc64" => Ok(Platform::LinuxPpc64),
-            "s390x" => Ok(Platform::LinuxS390X),
-            _ => Err(ConvertSubdirError::NoKnownCombination { platform, arch }),
-        },
-        "osx" => match arch.as_ref() {
-            "x86_64" => Ok(Platform::Osx64),
-            "arm64" => Ok(Platform::OsxArm64),
-            _ => Err(ConvertSubdirError::NoKnownCombination { platform, arch }),
-        },
-        "windows" => match arch.as_ref() {
-            "x86" => Ok(Platform::Win32),
-            "x86_64" => Ok(Platform::Win64),
-            "arm64" => Ok(Platform::WinArm64),
-            _ => Err(ConvertSubdirError::NoKnownCombination { platform, arch }),
-        },
-        _ => Err(ConvertSubdirError::NoKnownCombination { platform, arch }),
-    }?;
-    // Convert back to Platform string which should correspond to known subdirs
-    Ok(plat.to_string())
+    match arch.parse::<Arch>() {
```

Comment:
the previous function was not covering e.g. `emscripten-wasm32` so I modernized it a bit.